src/wallet/rpcwallet.cpp

```diff
@@ -30,10 +31,27 @@
 
 #include <univalue.h>
 
+static const std::string WALLET_ENDPOINT_BASE = "/v1/wallet/";
+
 CWallet *GetWalletForJSONRPCRequest(const JSONRPCRequest& request)
 {
-    // TODO: Some way to access secondary wallets
-    return vpwallets.empty() ? nullptr : vpwallets[0];
+    LOCK(cs_main); // for now, protect vpwallets via cs_main
+
+    if (request.URI.substr(0, WALLET_ENDPOINT_BASE.size()) == WALLET_ENDPOINT_BASE) {
+        // wallet endpoint was used
+        std::string requestedWallet = urlDecode(request.URI.substr(WALLET_ENDPOINT_BASE.size()));
+        for(CWallet* pwallet : vpwallets) {
+            if (pwallet->GetName() == requestedWallet) {
+                return pwallet;
+            }
+        }
+        // no wallet found with the given endpoint
+        throw JSONRPCError(RPC_WALLET_NOT_FOUND, "Wallet file does not exist or is not loaded");
+    }
+    else {
+        // default endpoint, use first wallet if possible
+        return vpwallets.empty() ? nullptr : vpwallets[0];
```

Comment:
Please see @ryanofsky's previous comment: https://github.com/bitcoin/bitcoin/pull/10650#discussion_r126928365

Returning a 'default' wallet is dangerous for users. If there is more than one wallet loaded and the RPC does not specify a wallet endpoint, then the call should fail.